runtime/vm/VMSnapshotImpl.cpp

```diff
@@ -714,6 +714,25 @@ VMSnapshotImpl::fixupClass(J9Class *clazz)
 	if (NULL != clazz->invokeCache) {
 		memset(clazz->invokeCache, 0, sizeof(UDATA) * clazz->romClass->invokeCacheCount);
 	}
+
+	omrthread_monitor_enter(_vm->memberNameListsMutex);
+	if (NULL != clazz->memberNames) {
+		J9InternalVMFunctions *vmFuncs = _vm->internalVMFunctions;
+		J9MemberNameListNode *node = clazz->memberNames;
+		J9VMThread *vmThread = currentVMThread(_vm);
+		clazz->memberNames = NULL;
+		while (NULL != node) {
+			J9MemberNameListNode *next = node->next;
+			/* The weak JNI ref must have been cleared by now. Otherwise, the
+			* MemberName is still live, and this class should be too.
+			*/
```

Comment:
This comment can be removed. This code will be executed while the snapshot is created. The snapshot is created at the end of the application, and all weak JNI refs should be cleared by then.